script.js

```diff
@@ -0,0 +1,178 @@
+const numArray = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
+const specialArray = [
+  "!",
+  "£",
+  "$",
+  "%",
+  "^",
+  "&",
+  "*",
+  "(",
+  ")",
+  "_",
+  "-",
+  "+",
+  "=",
+  "?",
+  "@",
+  "#",
+  "~",
+  "|",
+  "/",
+  ".",
+  ",",
+  "<",
+  ">",
+  '"',
+  ":",
+  " ",
+  ";",
+  "`",
+  "¬",
+];
+const lowerCaseArray = [
+  "a",
+  "b",
+  "c",
+  "d",
+  "e",
+  "f",
+  "g",
+  "h",
+  "i",
+  "j",
+  "k",
+  "l",
+  "m",
+  "n",
+  "o",
+  "p",
+  "q",
+  "r",
+  "s",
+  "t",
+  "u",
+  "v",
+  "w",
+  "x",
+  "y",
+  "z",
+];
+
+// convert lower case to uppercase array
+const upperCase = lowerCaseArray.join("").toUpperCase();
+const upperCaseArray = upperCase.split("");
+
+// get criteria
+// Function - prompt user to select password characteristics
+const getCriteria = function () {
+  // created a variable x which will hold the users numerical input
+  const x = prompt("Choose Password length");
+  // variable is converted to a number and passed into a new variable called passwordLength
+  const passwordLength = parseInt(x, 10);
+  // checks if the value passed is a number,
+  if (Number.isNaN(passwordLength)) {
+    // if it is not a number the return the following message
+    alert("Please enter a valid number");
+    // then break away
+    return;
+  }
+
+  // length requirement error handler
+  if (passwordLength < 8 || passwordLength > 128) {
+    // if number is of an incorrect length the return the following message
+    alert("Please enter a number between 8-128");
+    // then break away
+    return;
+  }
+
+  // variable assigning each criteria the user can choose from
+  // Confirmation box - pop up asking user for boolean data type in return
+  const lowerCase = confirm("Password to contain lowercase characters?");
+  const upperCase = confirm("Password to contain uppercase characters?");
+  const specialChar = confirm("Password to contain special characters?");
+  const numChar = confirm("Password to contain numbers?");
+  // created an object called 'attributes', holding all the variables the user will return
+  const attributes = {
+    passwordLength: passwordLength,
+    lowerCase: lowerCase,
+    upperCase: upperCase,
+    specialChar: specialChar,
+    numChar: numChar,
+  };
+  return attributes;
+};
+
+// function that generates the password according to user choices
+const generatePassword = function () {
+  // call on getCriteria function to use variables
+  const attributes = getCriteria();
+  // empty array that will hold the arrays the user wishes to use
+  const containerArray = [];
+  // if user has received error message finishes the function
+  if (!attributes) {
+    return;
+  }
+  // if the named attributes are selected, push into the empty array using the spread function
+  // container array goes first as thats the container we want to fill, push is the function to add,
+  // the ...is the spread function that we attach to the array we are forcing into a new array.
+
```

Comment:
These sort of comments, whilst very detailed, are not necessary. Try and condense some of your comments to help your code look cleaner.